### PR TITLE
#164797990 Implement Search Functionality

### PR DESCRIPTION
**Pull request by Akinmyde**
#### Description
This PR implement search functionality for the application

#### Type of change
- [x] Install lodash
- [x] design search view
- [x] write logic for search

#### How Has This Been Tested?
- [x] Unit testing

#### Checklist:
N/A

#### PT-ID
[#164797990](https://www.pivotaltracker.com/story/show/164797990)

#### Screenshots
Screenshot 2019-05-21 at 1 26 01 AM

#### Questions:
N/A

### CODE REVIEW

**Review by sojida — APPROVED** [review]

File: src/components/shared/Search/Search.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        authors,
+        tags,
+      ]);
+      const filteredResults = this.filterResultByName(source);
+
+      return this.setState({
+        isLoading: false,
+        results: filteredResults,
+      });
+    }, 500);
```
Comment (rotimi-babalola) [inline src/components/shared/Search/Search.jsx:89]: Are we waiting 500 milliseconds now? @Akinmyde

Comment (Akinmyde) [reply src/components/shared/Search/Search.jsx:89]: Yes, I found out that 3000 milliseconds will be a lot

**Review by walsamlee — APPROVED** [review]

File: src/components/shared/Search/Search.jsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        id: tags.article.id,
+        title: tags.article.title,
+        description: Parser(tags.article.body.substring(0, 55)),
+        image: tags.article.author.image_url,
+        price: tags.keyword,
+      });
+    });
+    authorFilter.forEach(author => {
+      newAuthors.push({
+        title: author.first_name + author.last_name,
```
Comment (rotimi-babalola) [inline src/components/shared/Search/Search.jsx]: Use ES6 string templates for concatenation

File: src/components/shared/Search/Search.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import { searchArticle } from '../../../actions/search.action';
+import './search.scss';
+
+class SearchComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: false,
+      value: '',
+      getResults: {
```
Comment (rotimi-babalola) [inline src/components/shared/Search/Search.jsx]: Let's give this a better name

File: src/components/shared/Search/Search.jsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        articles: {},
+        authors: {},
+        tags: {},
```
Comment (rotimi-babalola) [inline src/components/shared/Search/Search.jsx]: The initial values should be an empty array

**Pull request merged**